tests/lambda.rs

```diff
@@ -39,15 +39,32 @@ struct LambdaAnalysis;
 #[derive(Debug)]
 struct Data {
     free: HashSet<Id>,
-    constant: Option<Lambda>,
+    constant: Option<(Lambda, PatternAst<Lambda>)>,
 }
 
-fn eval(egraph: &EGraph, enode: &Lambda) -> Option<Lambda> {
-    let x = |i: &Id| egraph[*i].data.constant.clone();
+fn eval(egraph: &EGraph, enode: &Lambda) -> Option<(Lambda, PatternAst<Lambda>)> {
+    let x = |i: &Id| egraph[*i].data.constant.as_ref().map(|c| &c.0).clone();
+    let pat = |i: &Id| egraph[*i].data.constant.as_ref().map(|c| &c.1).clone();
     match enode {
-        Lambda::Num(_) | Lambda::Bool(_) => Some(enode.clone()),
-        Lambda::Add([a, b]) => Some(Lambda::Num(x(a)?.num()? + x(b)?.num()?)),
-        Lambda::Eq([a, b]) => Some(Lambda::Bool(x(a)? == x(b)?)),
+        Lambda::Num(n) => Some((
+            enode.clone(),
+            format!("{}", n).parse().unwrap(),
+        )),
+        Lambda::Bool(b) => 
+            Some((
+            enode.clone(),
+            format!("{}", b).parse().unwrap(),)),
+            
+        Lambda::Add([a, b]) =>
+            Some((
+            Lambda::Num(x(a)?.num()? + x(b)?.num()?),
+            format!("(+ {} {})", pat(a)?, pat(b)?).parse().unwrap(),
```

Comment:
There's a very subtle problem here- we would like to insert simply pattern corresponding to (+ constant constant)
But the class analysis may not have unioned the constant with the class yet due to delayed unions
So we have to recursively build the pattern from the children, which is bad. Not sure how to fix this yet